pkg/etcd/etcd.go

```diff
@@ -43,7 +44,9 @@ import (
 )
 
 type Server struct {
-	Dir string
+	Dir        string
+	config     *embed.Config
+	clientInfo *ClientInfo
 }
```

Comment:
if this struct was called `Config`, and `Init` was called `NewServer` and it was returning a `Server` struct, it would match the other structure much better. Also `ClientInfo()` then wouldn't need to check whether `Init()` has been called.